src/Components/events/DetailEvent.js

```diff
@@ -20,26 +16,20 @@ const DetailEvent = () => {
   const eventId = useParams().id;
   const location = useLocation();
   const userId = location.state.userId;
+  const eventsList = location.state.eventsList;
   const [goToCreateNewEvent, setGoToCreateNewEvent] = useState(false);
   const [goToDashboard, setGoToDashboard] = useState(false);
   const [goToEditEvent, setGoToEditEvent] = useState(false);
   const [eventToEdit, setEventToEdit] = useState("");
-  const [eventsList, setEventsList] = [];
-
-  useEffect(() => {
-    async function getEvents() {
-      const eventsFound = await getEventsFromServer();
```

Comment:
create a function called `getEventFromServer`, call this function passing the `eventId`, like: `getEventFromServer(eventId)`.  Inside that function, get all events and filter them by `eventId` and return the event we need.